src/frontends/tensorflow/src/decoder_proto.cpp

```diff
@@ -108,9 +108,10 @@ ov::Any DecoderProto::get_attribute(const std::string& name) const {
         if (tf_shape.unknown_rank()) {
             return ov::PartialShape::dynamic();
         }
-        std::vector<ov::Dimension> dims(tf_shape.dim_size());
-        for (int i = 0; i < tf_shape.dim_size(); ++i) {
-            dims[i] = tf_shape.dim(i).size();
+        size_t shape_rank = static_cast<size_t>(tf_shape.dim_size());
+        std::vector<ov::Dimension> dims(shape_rank);
+        for (size_t i = 0; i < shape_rank; ++i) {
+            dims[i] = static_cast<ov::Dimension>(tf_shape.dim(i).size());
```

Comment:
Casting tf dim value to `ov::Dimension` is weird. Can you call `ov::Dimension` constructor here?